packages/components/src/InputNumber/InputNumber.rebuilt.module.css

```diff
@@ -0,0 +1,251 @@
+  --field--background-color: var(--color-disabled--secondary);
+  --field--border-color: var(--color-border);
+}
+
+/* override iOS default styling for disabled input */
+.disabled :disabled {
+  -webkit-text-fill-color: var(--field--value-color);
+  opacity: 1;
+}
+
+.description {
+  margin-top: var(--space-smaller);
+}
+
+.fieldError {
+  align-items: center;
+  display: flex;
+  gap: var(--space-smaller);
+  padding: var(--space-smaller);
+  padding-left: 0;
+}
+
+.small {
+  --field--padding-left: calc(var(--space-base) - var(--space-smaller));
+  --field--padding-right: calc(var(--space-base) - var(--space-smaller));
+  --field--padding-top: var(--space-small);
+  --field--padding-bottom: var(--space-small);
+  --field--height: calc(var(--space-larger) + var(--space-smaller));
+  --postfix-right: calc(var(--space-small) + var(--space-smaller));
+}
+
+.large {
+  --field--padding-left: var(--space-large);
+  --field--padding-right: var(--space-large);
+  --field--height: calc(var(--space-extravagant));
+  --postfix-right: var(--space-large);
+}
+
+.center {
+  --field--textAlign: center;
+}
+
+.right {
+  --field--textAlign: right;
+}
+
+/**
+ * Input Wrapper
+ **/
+.inputWrapper {
+  display: flex;
+  flex: 1;
+  position: relative;
+  /* Ensure input stays with the rest of the element stack */
+  z-index: var(--elevation-default);
+}
+
+.childrenWrapper {
+  position: relative;
+  flex: 1;
+}
+
+/**
+ * Input
+ **/
+.input {
+  position: relative;
+  z-index: var(--field--base-elevation);
+  width: 100%;
+  padding-top: var(--field--padding-top);
+  padding-bottom: var(--field--padding-bottom);
+  padding-left: var(--field--padding-left);
+  padding-right: var(--field--padding-right);
+  border: none;
+  border-radius: var(--radius-base);
+  color: inherit;
+  font-family: inherit;
+  font-size: inherit;
+  line-height: var(--field--value-lineHeight);
+  text-align: var(--field--textAlign);
+  background: transparent;
+  appearance: none;
+}
+
+.input:focus {
+  outline: none;
+}
+
+/* DO WE NEED TO SUPPORT AUTOFILL FOR INPUT NUMBER?? */
+/* .input:-webkit-autofill,
+.input:-webkit-autofill:hover,
+.input:-webkit-autofill:focus,
+.input:-webkit-autofill:active {
+  -webkit-box-shadow: 0 0 0 30px var(--color-surface) inset !important;
+} */
+
+/**
+ * Label
+ **/
+.label {
+  position: absolute;
+  top: var(--field--placeholder-offset);
+  z-index: var(--field--label-elevation);
+  width: 100%;
+  padding-left: var(--field--padding-left);
+  padding-right: var(--field--padding-right);
+  overflow: hidden;
+  color: var(--field--placeholder-color);
+  line-height: var(--field--value-lineHeight);
+  text-align: var(--field--textAlign);
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  pointer-events: none;
+  transform: translateY(var(--field--placeholder-transform));
+}
+/* When not hiding the label, animate the transition to the miniLabel */
+.inputWrapper:not(.hideLabel) .label {
+  transition: all var(--timing-quick);
+}
+/**
```

Comment:
Some of the rule logic gets a bit complex here. I promised @ZakaryH I would address that in a followup. For now, I wanted to get this into review but I'll work on the followup alongside this. (It's a bit finicky to make sure all of the logic gets correctly preserved.)